src/pool.rs

```diff
@@ -348,10 +361,7 @@ impl ConnectionPool {
                         // Don't leave a bad connection in the pool.
                         server.mark_bad();
 
-                        self.ban(address, shard);
-                        self.stats.client_disconnecting(process_id, address.id);
-                        self.stats
-                            .checkout_time(now.elapsed().as_micros(), process_id, address.id);
+                        self.ban(address, shard, process_id);
```

Comment:
Could you add a `debug!` a little bit above so we know that the health check is still being performed? It's nice to trace the execution of things when debugging things. I tend to organize the log levels as follows:

- `info`: good to know under normal operations, e.g. client connected, client disconnected, etc.
- `warn`: need to know, something might be wrong.
- `debug`: good to know if I'm trying to understand what's really happening underneath.
- `trace`: I need to see almost every interaction and byte going through because I think something is terribly wrong.

Reply:
So right above it there's already an error! log which says the replica is being banned. Does that work?

Reply:
I was referring to the health check being issued, not the ban, but GitHub doesn't let me comment on random file lines i think.

Reply:
Now that we don't always health check i think it's important to log when we do.